src/pool/cli.c

```diff
@@ -1305,12 +1428,20 @@ pool_query_cb(tse_task_t *task, void *data)
 		D_GOTO(out, rc);
 	}
 
+	ranks_arg = arg->dqa_ranks ? arg->dqa_ranks : &ranks;
 	rc = process_query_reply(arg->dqa_pool, map_buf,
 				 out->pqo_op.po_map_version,
 				 out->pqo_op.po_hint.sh_rank,
 				 &out->pqo_space, &out->pqo_rebuild_st,
-				 arg->dqa_tgts, arg->dqa_info,
+				 ranks_arg /* arg->dqa_ranks */, arg->dqa_info,
```

Comment:
Debug leftover, fixed in latest push. Added ranks_arg approach here since the tools (e.g., daos) do not yet pass a rank list, and I wanted to see (through client logging) evidence of how various exclude/drain/reintegrate scenarios affected list population.